modules/videogrid-workflowoperation/src/main/java/org/opencastproject/workflow/handler/videogrid/VideoGridWorkflowOperationHandler.java

```diff
@@ -0,0 +1,1007 @@
+          if (trackDimension == null) {
+            throw new WorkflowOperationException("One of the source video tracks did not contain a valid video stream or dimension");
+          }
+          videoInfo.aspectRatioHeight = trackDimension.getHeight();
+          videoInfo.aspectRatioWidth = trackDimension.getWidth();
+          // "StartTime" is calculated later. It describes how far into the video the next section starts.
+          // (E.g. If webcam2 is started 10 seconds after webcam1, the startTime for webcam1 in the next section is 10)
+          videoInfo.startTime = 0;
+
+          logger.info("Video information: Width: {}, Height {}, StartTime: {}", videoInfo.aspectRatioWidth,
+                  videoInfo.aspectRatioHeight, videoInfo.startTime);
+
+          events.add(new StartStopEvent(true, getTrackPath(track), beginInMs, videoInfo));
+          events.add(new StartStopEvent(false, getTrackPath(track), beginInMs + durationInMs, videoInfo));
+
+        }
+      }
+    }
+
+    // No events? Skip
+    if (events.isEmpty()) {
+      logger.warn("Could not generate sections from given SMIL catalogue for tracks in given flavor, skipping ...");
+      return createResult(mediaPackage, WorkflowOperationResult.Action.SKIP);
+    }
+
+    // Sort by timestamps ascending
+    Collections.sort(events);
+
+    // Create an edit decision list
+    List<VideoEdlPart> videoEdl = new ArrayList<VideoEdlPart>();
+    HashMap<String, StartStopEvent> activeVideos = new HashMap<>();   // Currently running videos
+
+    // Define starting point
+    VideoEdlPart start = new VideoEdlPart();
+    start.timeStamp = finalStartTime;
+    videoEdl.add(start);
+
+    // Define mid-points
+    for (StartStopEvent event : events) {
+      if (event.start) {
+        logger.info("Add start event at {}", event.timeStamp);
+        activeVideos.put(event.filename, event);
+      } else {
+        logger.info("Add stop event at {}", event);
+        activeVideos.remove(event.filename);
+      }
+      videoEdl.add(createVideoEdl(event, activeVideos));
+    }
+
+    // Define ending point
+    VideoEdlPart endVideo = new VideoEdlPart();
+    endVideo.timeStamp = finalEndTime;
+    endVideo.nextTimeStamp = finalEndTime;
+    videoEdl.add(endVideo);
+
+    // Pre processing EDL
+    for (int i = 0; i < videoEdl.size() - 1; i++) {
+      // For calculating cut lengths
+      videoEdl.get(i).nextTimeStamp = videoEdl.get(i + 1).timeStamp;
+    }
+
+    // Create ffmpeg command for each section
+    List<List<String>> commands = new ArrayList<>();
+    for (VideoEdlPart edl : videoEdl) {
+      // A too small duration will result in ffmpeg producing a faulty video, so avoid any section smaller than 50ms
+      if (edl.nextTimeStamp - edl.timeStamp < 50) {
+        logger.info("Skipping {}-length edl entry", edl.nextTimeStamp - edl.timeStamp);
+        continue;
+      }
+      // Create command for section
+      commands.add(compositeSection(layoutArea, edl));
+    }
```

Comment:
It would be nice if this minlen parameter could be set as a workflow parameter. I could even imagine setting it to something like 1 sec, so the video does not change that often in case a lot of people join or leave

Reply:
Interesting proposal, I can see the benefit. Although it could perhaps also turn into an unintended error source, as currently the time period covered by the skipped edl entry is lost completely. So if entries are skipped often or the minlen is very high, the final video could become substantially shorter than anticipated.

Reply:
Honestly I'd prefer a solution to this problem that avoids skipping edl entries altogether (if their length is >0 of course), but I'm frankly don't know how such a solution would look like.

Reply:
One option, might be, to pad the videos, which would cause these issue and would qualify for an extra PR. So for now, just leave it as it is.